src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -21,28 +21,38 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/login";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    boolean loggedIn = userService.isUserLoggedIn();
+    String urlToRedirect = "/search.html";
 
-      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    JSONObject userDetails = new JSONObject();
+    userDetails.put("loggedIn", loggedIn);
+    if (loggedIn) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      userDetails.put("userEmail", userEmail);
+      String logoutURL = userService.createLogoutURL(urlToRedirect);
+      userDetails.put("url", logoutURL);
+      int start = userEmail.indexOf('@');
+      int end = userEmail.indexOf('.');
+      String schoolName;
+      if (start == -1 || end == -1) { // This means that @ or . are not in userEmail.
+        schoolName = "Sorry, you have entered an invalid email.";
```

Comment:
The better practice here is to just not set the school name

Reply:
Makes sense, will negate this OR statement and only set school name if the right conditions are met